sqs-lambda-ddb-cdk-ts/README.md

```diff
@@ -0,0 +1,129 @@
+* [AWS CLI](https://docs.aws.amazon.com/cli/latest/userguide/install-cliv2.html) installed and configured
+* [Git Installed](https://git-scm.com/book/en/v2/Getting-Started-Installing-Git)
+* [Node and NPM installed](https://docs.npmjs.com/downloading-and-installing-node-js-and-npm/)
+* [AWS Cloud Development Kit](https://docs.aws.amazon.com/cdk/v2/guide/getting_started.html) (AWS CDK) installed and bootstrapped
+
+## Deployment Instructions
+
+1. Create a new directory on your machine, navigate to that directory and clone this repository.
+
+2. Navigate to the directory of this repository:
+
+```
+cd sqs-lambda-ddb-cdk-ts
+```
+
+3. Install Node Packages:
+
+```
+npm install
+```
+
+4. From the command line, use AWS CDK to synthesize an AWS CloudFormation Template:
+
+```
+cdk synth
+```
+
+5. From the command line, use AWS CDK to deploy the AWS resources:
+
+```
+cdk deploy
+```
+
+## Testing
+
+Use the provided test script to send messages to the SQS queue. We will simulate a scenario where N messages are sent to the queue per second.
+
+1. Navigate to the test-scripts directory and open up the file named push-sqs-messages.js
+
+```
+cd test-scripts
+```
+
+2. In the script, replace 'YOUR_REGION' and 'YOUR_SQS_QUEUE_URL' with your desired AWS Region and SQS Queue URL respectively
+
+3. Using the command line, run the script and specify an integer for N (5 is a sufficient number for this demo) and for T(1 is sufficient for this demo). Keep this script running in the background until later in step 10. You should begin to see continuous messages in your terminal saying "Batch of N items pushed to SQS at ...".
+
+```
+node push-sqs-messages.js N T
+```
+
+4. At this point, the Lambda function will start inserting items into the DynamoDB table. After a couple of minutes, you should notice the DLQ filling up with messages since DynamoDB will start throttling the requests.
+
+5. Using Cloudwatch Log Insights, run a query on the Lambda function's Log Group to ensure that the requests are being throttled. Feel free to use the new Query Generator feature or simply run the following query:
+
+```
+filter @message like /ProvisionedThroughputExceededException/
+```
+
+
+
+6. Once you have confirmed that DynamoDB is throttling write requests, go back to the [config](./lib/config/tables.config.ts) file in repo and lower the `lambdaReservedConcurrency` property to a value less than or equal to the current writeCapacity value of the table (1 is a sufficient value for this demo). Increase the `retryAttempts` property value to a high value (20 should be sufficient for this demo) to give Lambda time to keep up with any subsequent throttling events.
```

Comment:
The `writeCapacity` of the table is already at 1, how should the `lambdaReservedConcurrency` be reduced even more?